### PR TITLE
Uplift

### PR DESCRIPTION
**Pull request by ivyONS**
Bring up to speed with most changes from development in private repo

### CODE REVIEW

File: fast-api/main_api.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+app = FastAPI(
+    title="ONS StatsChat API",
+    description=(
+        "Read more in [blog post]"
+        + "(https://datasciencecampus.ons.gov.uk/using-large-language-models-llms-to-improve-website-search-experience-with-statschat/)"  # noqa: E501
+        + " or see [the code repository]"
+        + "(https://github.com/datasciencecampus/statschat-app). "
+        + "Frontend UI available "
+        + "[https://statschat-275h3xjs4a-nw.a.run.app]"
+        + "(https://statschat-275h3xjs4a-nw.a.run.app)."
```
Comment (daffidwilde) [inline fast-api/main_api.py:42]: Is this behind some sort of authentication service now? If not, do we care that this URL is public?

File: flask-app/templates/statschat.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+{% if results %}
+
+<section class="search__summary" role="contentinfo" aria-label="Search summary" style = "display:inline;">
+    <div class="secondary-nav " style = "display:inline;justify-content:right;">
+        <i class="glyphicon glyphicon-thumbs-up feedback-up" style="font-size:24px;color:grey;" id="thumbs-up-ico"></i>
+        <i class="glyphicon glyphicon-thumbs-down feedback-down" style="font-size:24px;color:grey;" id="thumbs-down-ico"></i>
+        <input type="text"  placeholder="Enter feedback comment." style="display:none;" id="feedback-send-input">
+        <span class="feedback-send" style="display:none;" id="feedback-send-icon"> Send</span>
+        <script src="https://code.jquery.com/jquery-3.6.0.js" integrity="sha256-H+K7U5CnXl1h5ywQfKtSj8PCmoN9aaq30gDh27Xc0jk=" crossorigin="anonymous"></script>
```
Comment (daffidwilde) [inline flask-app/templates/statschat.html:13]: Is this a SHA-256 key?

File: notebooks/load_llm_models_docker.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# this script is to be run when docker image is constructed so that the required models
+# would be included in the image and don't need to be donwloaded on each instance start
```
Comment (daffidwilde) [inline notebooks/load_llm_models_docker.py:2]:
```suggestion
# would be included in the image and don't need to be downloaded on each instance start
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+dev = [
+    "ipykernel==6.23.2",
+    "pre-commit==3.3.3",
+    "statschat[test]",
+    "statschat[backend]",
+    "statschat[frontend]",
+]
+
+[project.urls]
+homepage = "https://github.com/datasciencecampus/statschat"
```
Comment (daffidwilde) [inline pyproject.toml:49]:
```suggestion
homepage = "https://github.com/datasciencecampus/statschat-app"
```

**Review by daffidwilde — COMMENTED** [review]
Hi @ivyONS - just a couple of small points to assess. Once they're resolved, I would be happy to do another review and approve.

File: flask-app/templates/statschat.html (hunk 2 above)
Comment (ivyONS) [reply flask-app/templates/statschat.html:13]: it's the expected hash if the file to check that it got the correct/uncorrupted one from the 3rd party host, I think.

File: fast-api/main_api.py (hunk 1 above)
Comment (ivyONS) [reply fast-api/main_api.py:42]: ip filter, but better to remove

**Pull request closed without merging**